app/src/main/java/com/android/sample/model/recipe/MealDBRecipesRepository.kt

```diff
@@ -31,6 +32,22 @@ import org.json.JSONObject
  */
 class MealDBRecipesRepository(private val client: OkHttpClient) : RecipesRepository {
 
+  override fun getNewUid(): String {
+    throw UnsupportedOperationException(UNSUPPORTED_MESSAGE)
+  }
+
```

Comment:
Nice work here! I can see that the updates primarily handle unsupported methods and add a limit parameter to `searchByCategory`.

**Minor Suggestions**:
1. **Comment for Unsupported Methods**: Consider adding a brief comment above each unsupported method to clarify why it’s unsupported in `MealDBRecipesRepository`. It might help future developers quickly understand the reason.

Overall, it’s looking solid and consistent.

Reply:
Done!